File: examples/transforms/approval_down_guess_manual.py
#!/usr/bin/env python3
import json, sys
y = json.load(sys.stdin)
if y.get("approved") is True:
    print(json.dumps({"approval": "manual"}))
elif y.get("approved") is False:
    print(json.dumps({"approval": "rejected"}))
else:
    print("unsupported approved value", file=sys.stderr)
    sys.exit(12)

File: examples/transforms/approval_down_guess_manual.js
#!/usr/bin/env node
"use strict";

const input = readJson();

if (input.approved === true) {
  writeJson({ approval: "manual" });
} else if (input.approved === false) {
  writeJson({ approval: "rejected" });
} else {
  process.stderr.write("unsupported approved value\n");
  process.exitCode = 12;
}

function readJson() {
  return JSON.parse(require("node:fs").readFileSync(0, "utf8"));
}

function writeJson(value) {
  process.stdout.write(`${JSON.stringify(value)}\n`);
}
